I didn't finish running nix-review before killing it, but it looks like there might be an issue where the new release doesn't support python2. Since some packages are depending on toolz for python2, it might be a good idea to split it up into a `default.nix` and a 2.nix, like [pandas](https://github.com/NixOS/nixpkgs/tree/master/pkgs/development/python-modules/pandas) has done, so that the python27Packages version can use the older release of toolz.

```
$ nix-review pr 101363
$ git -c fetch.prune=false fetch --force https://github.com/NixOS/nixpkgs master:refs/nixpkgs-review/0 pull/101363/head:refs/nixpkgs-review/1
remote: Enumerating objects: 16, done.
remote: Counting objects: 100% (16/16), done.
remote: Compressing objects: 100% (9/9), done.
remote: Total 19 (delta 8), reused 7 (delta 7), pack-reused 3
Unpacking objects: 100% (19/19), 34.88 KiB | 871.00 KiB/s, done.
From https://github.com/NixOS/nixpkgs
   066bf7dd5d1..c69407d6060  master     -> refs/nixpkgs-review/0
$ git worktree add /home/mcgibbon/.cache/nixpkgs-review/pr-101363-2/nixpkgs c69407d60608b239a8f90f04e66fc8f9aa91c7db
Preparing worktree (detached HEAD c69407d6060)
Updating files: 100% (23067/23067), done.
HEAD is now at c69407d6060 gallery-dl: 1.15.1 -> 1.15.2 (#101977)
$ git merge --no-commit aaf0227a3a7dcfeea2a58dacc354f1715cfe0042
Automatic merge went well; stopped before committing as requested
$ nix build --no-link --keep-going --option build-use-sandbox relaxed -f /home/mcgibbon/.cache/nixpkgs-review/pr-101363-2/build.nix
warning: ignoring the user-specified setting 'sandbox', because it is a restricted setting and you are not a trusted user
builder for '/nix/store/wa9w6p88j874rnn043h3jbl32p4myigw-python2.7-toolz-0.11.1.drv' failed with exit code 1; last 10 log lines:
  adding 'toolz-0.11.1.dist-info/top_level.txt'
  adding 'toolz-0.11.1.dist-info/RECORD'
  removing build/bdist.linux-x86_64/wheel
  Finished executing setuptoolsBuildPhase
  installing
  Executing pipInstallPhase
  /build/toolz-0.11.1/dist /build/toolz-0.11.1
  DEPRECATION: Python 2.7 reached the end of its life on January 1st, 2020. Please upgrade your Python as Python 2.7 is no longer maintained. pip 21.0 will drop support for Python 2.7 in January 2021. More details about Python 2 support in pip, can be found at https://pip.pypa.io/en/latest/development/release-process/#python-2-support
  Processing ./toolz-0.11.1-py2-none-any.whl
  ERROR: Package 'toolz' requires a different Python: 2.7.18 not in '>=3.5'
cannot build derivation '/nix/store/fja1cmxfc94wxcszlfpjh5i45rsizafn-python2.7-strategies-0.2.3.drv': 1 dependencies couldn't be built
builder for '/nix/store/22jp43i5jabigqbya64j2fwv7542wmzp-python3.7-intake-0.6.0.drv' failed with exit code 1; last 10 log lines:
      result = self.orig_method(*args, **kwargs)
  
  intake/gui/tests/test_gui.py::test_gui_open_plot_panel
    /build/intake-0.6.0/intake/gui/tests/test_gui.py:63: UserWarning: direct access to the entry is deprecated
      assert gui.source.plot.source.entry == sources1[0]
  
  -- Docs: https://docs.pytest.org/en/latest/warnings.html
  =========================== short test summary info ============================
  FAILED intake/catalog/tests/test_caching_integration.py::test_list_of_files
  = 1 failed, 420 passed, 13 skipped, 14 deselected, 13 warnings in 276.76s (0:04:36) =
[6/27/64 built (2 failed)] building python3.8-clifford-1.3.1 (pytestCheckPhase): test_g3c_CUDA.py sssssssssssssss                                         [ 32%]
```

--- a/pkgs/development/python-modules/toolz/2.nix
+++ b/pkgs/development/python-modules/toolz/2.nix
@@ -1,0 +1,28 @@
+{ lib
+, buildPythonPackage
+, fetchPypi
+, nose
+}:
+
+buildPythonPackage rec {
+  pname = "toolz";
+  version = "0.10.0";
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "08fdd5ef7c96480ad11c12d472de21acd32359996f69a5259299b540feba4560";
+  };
+
+  checkInputs = [ nose ];
+
+  checkPhase = ''
+    nosetests toolz/tests
+  '';
+
+  meta = with lib; {
+    homepage = "https://github.com/pytoolz/toolz";
+    description = "List processing tools and functional utilities";
+    license = licenses.bsd3;
+    maintainers = with maintainers; [ fridh ];
+  };
+}

--- a/pkgs/development/python-modules/toolz/default.nix
+++ b/pkgs/development/python-modules/toolz/default.nix
@@ -6,18 +6,16 @@
 
 buildPythonPackage rec {
   pname = "toolz";
-  version = "0.10.0";
+  version = "0.11.1";
 
   src = fetchPypi {
     inherit pname version;
-    sha256 = "08fdd5ef7c96480ad11c12d472de21acd32359996f69a5259299b540feba4560";
+    sha256 = "1grz3zvw5ixwqqlbv0n7j11mlcxb66cirh5i9x9zw8kqy0hpk967";
   };
 
   checkInputs = [ nose ];
 
   checkPhase = ''
-    # https://github.com/pytoolz/toolz/issues/357
-    rm toolz/tests/test_serialization.py
     nosetests toolz/tests
   '';
 

--- a/pkgs/top-level/python-packages.nix
+++ b/pkgs/top-level/python-packages.nix
@@ -7175,7 +7175,10 @@ in {
 
   tomlkit = callPackage ../development/python-modules/tomlkit { };
 
-  toolz = callPackage ../development/python-modules/toolz { };
+  toolz = if isPy3k then
+    callPackage ../development/python-modules/toolz { }
+  else
+    callPackage ../development/python-modules/toolz/2.nix { };
 
   toposort = callPackage ../development/python-modules/toposort { };
 

